Cauli/Florets/Inspector/Record List/InspectorTableViewController.swift

```diff
@@ -111,6 +114,7 @@ extension InspectorTableViewController: UISearchResultsUpdating {
         let searchString = searchController.searchBar.text ?? ""
         dataSource.filterString = searchString
         tableView.reloadData()
+        loadAdditionalRecords()
```

Comment:
This would now load more data even if there is enough data visible. For example if one is typing "https://" and that will be part of every single request, this logic will load 8 more pages of records even though no more additional records would be needed.

Admittedly, this is pretty nitpicky. I'll think about if I have a better idea than that. If non of us has, I would be fine to merge it as it is.

Reply:
Two ways to fix that:

1. Calculate how many rows fit on the screen and only load that many rows. 

2. Move filtering of the record list to the storage, so that we can always load a new, filtered result set instead of filtering the current results. 

I like 2 better, but I am not sure if it’s not too much for this issue.

Reply:
I think you are correct. We should rather change how the Storage works and that's not part of this scope. There are quite some more tasks to tackle in that context.

Reply:
I think for now I would love to change the `loadAdditionalRecords()` to a `loadAdditionalRecordsIfNeeded()` with the following implementation. What do you think about that?

```swift
    fileprivate func loadAdditionalRecordsIfNeeded() {
        let distanceToBottom = tableView.contentSize.height - tableView.frame.height - scrollView.contentOffset.y
        guard !scrolledToEnd, !isLoading, distanceToBottom < 100 else { return }
        isLoading = true
        let newRecords = cauli.storage.records(InspectorTableViewController.recordPageSize, after: dataSource.items.last)
        guard !newRecords.isEmpty  else {
            isLoading = false
            scrolledToEnd = true
            return
        }
        
        dataSource.append(records: newRecords, to: tableView) { [weak self] _ in
            self?.isLoading = false
        }
    }
```